src/drivers/distance_sensor/leddar_one/LeddarOne.cpp

```diff
@@ -0,0 +1,300 @@
+/****************************************************************************
+ *
+ *   Copyright (C) 2017-2019 Intel Corporation. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#include "LeddarOne.hpp"
+
+#include <stdlib.h>
+#include <string.h>
+
+LeddarOne::LeddarOne(const char *serial_port, uint8_t device_orientation):
+	ScheduledWorkItem(MODULE_NAME, px4::serial_port_to_wq(serial_port)),
+	_px4_rangefinder(0 /* device id not yet used */, ORB_PRIO_DEFAULT, device_orientation)
```

Comment:
@dagar would this be a good opportunity to make the device ID usable?

Reply:
All that's really missing is to add them here. https://github.com/PX4/Firmware/blob/master/src/drivers/drv_sensor.h

I was going to get it cleaned up a bit to try and have a dedicated range for the existing sensors, but it doesn't really matter.

Reply:
What if you have multiple sensors of the same type? How do you distinguish their ID's? And how do you match the ID with the uORB instance you want to use?

Reply:
The device ID contains the type (drv_sensor.h), bus, and address. The orb message contains the device id.

There's a helper to decode them that's used in listener. For example `listener sensor_accel` on fmu-v4 shows 2 accels with type 0x37 (DRV_ACC_DEVTYPE_ICM20602) and 0x16 (DRV_ACC_DEVTYPE_MPU9250).

<img width="384" alt="Screen Shot 2019-12-31 at 10 44 34 AM" src="https://user-images.githubusercontent.com/84712/71626329-93b14080-2bba-11ea-84c5-812af1a95bcc.png">